survey_auto_code/survey_auto_code.py

```diff
@@ -0,0 +1,143 @@
+        CleaningPlan("Age (Text) - mcf_demog", "age_clean", "Age",
+                     # TODO:: set age cleaner once coding scheme is ready
+                     #swahili.DemographicCleaner.clean_age),
+                     None, None),
+        CleaningPlan("Work (Text) - mcf_demog", "work_clean", "Work", None,
+                     "Scheme-12be1d8f34eb"),
+        CleaningPlan("Training (Text) - mcf_demog", "training_clean", "Training",
+                     None, "Scheme-8f0794281bb1"),
+    ]
+
+    # Load phone number UUID table
+    with open(phone_uuid_table_path, "r") as f:
+        phone_uuids = PhoneNumberUuidTable.load(f)
+
+    # Load data from JSON file
+    with open(json_input_path, "r") as f:
+        data = TracedDataJsonIO.import_json_to_traced_data_iterable(f)
+
+    # Filter out test messages sent by AVF
+    contacts = [td for td in data if not td.get("test_run", False)]
+
+    # Mark missing entries in the raw data as true missing
+    for td in data:
+        missing = dict()
+        for plan in cleaning_plan:
+            if plan.raw_field not in td:
+                missing[plan.raw_field] = Codes.TRUE_MISSING
+        td.append_data(missing, Metadata(user, Metadata.get_call_location(), time.time()))
+
+    # Exclude missing data
+    for plan in cleaning_plan:
+        data = [td for td in data if td[plan.raw_field] not in {Codes.TRUE_MISSING, Codes.SKIPPED, Codes.NOT_LOGICAL}]
+
+    # Clean all responses, add MessageID and Labels
+    for td in data:
+        cleaned = dict()
+        message_id = dict()
+        labels = dict()
+        for plan in cleaning_plan:
+            hash_object = hashlib.sha256()
+            hash_object.update(td[plan.raw_field].encode('utf-8'))
+            message_id_string = hash_object.hexdigest()
+            message_id_key = "{} MessageID".format(plan.raw_field)
+            message_id[message_id_key] = message_id_string
+            labels_key = "{} Labels".format(plan.raw_field)
+            labels[labels_key] = []
+            if plan.cleaner is not None:
+                label = dict()
+                cleaned[plan.clean_field] = plan.cleaner(td[plan.raw_field])
+                code_id = CODE_IDS[plan.scheme_id][cleaned[plan.clean_field]]
+                origin = {"OriginType":"Automatic","OriginID": "https://github.com/AfricasVoices/Project-MCF/pull/7", "Name": "survey_auto_code", "Metadata": {}}
+                label["Checked"] = "false"
+                label["SchemeID"] = plan.scheme_id
+                label["CodeID"] = code_id
+                label["DateTimeUTC"] = time.time()
+                label["Origin"] = origin
+                labels[labels_key].append(label)
+        td.append_data(cleaned, Metadata(user, Metadata.get_call_location(), time.time()))
+        td.append_data(message_id, Metadata(user, Metadata.get_call_location(), time.time()))
+        td.append_data(labels, Metadata(user, Metadata.get_call_location(), time.time()))
```

Comment:
These feel like three separate stages. I wonder if it would be cleaner to do each in turn i.e. set the message id, then clean, then set the labels. Maybe check with Luke